### ISSUE TITLE
Formie 2 inactive pages still loaded in Page-Reload mode

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#1522]
### Question

Maybe after updating from Formie 1 to 2, I uncounted an issue that I think I didn't have previously:

- I'm using a workaround to check and replace data from the current page
- What I found is that it was replacing content from another page that the active one
- The problem is that my form is setup using the _Page-Reload_ option
- However, I can see in the DOM that every `.fui-page` of the form are loaded, while useless (expect the active one)

Why are all `.fui-page` loaded in Page-Reload mode?
Is there a way to prevent this, in order to only load the current active page?

**Comment by engram-design** [comment]
This is correct behaviour for Formie 2 to load all pages even for Page Reload. The main reason for this is to address conditional field behaviour. For example, if you're on page 3, and some logic relies on content from page 1 or 2, the front-end will need to know about those values in order to evaluate conditions.

I'll agree that it's not _great_ that the entire form (multiple pages) submits on every page submit, but I've yet to come up with a better solution to handle conditions in this fashion.

**Comment by engram-design** [comment]
I know this was for Formie 2, but with Formie 3 and now Formie 4 on the horizon with no change in this behavior, I'll probably close this.

As I mentioned, it really comes down to missing out of a few things, like cross-page conditions, or JS modules that read the full form.

On Page-Reload, after someone completes page 1 and lands on page 2, page 1’s inputs are still in the DOM (populated from the saved submission). Conditions on page 2 that reference page 1 rely on that.

If we dropped inactive pages from the HTML, we'd need a replacement value source — a JS value registry, hidden transport fields, or server-only evaluation on every keystroke — before cross-page conditions would keep working live.

I will say that in Formie 4, Ajax submission has been improved to the current page's payload being sent to the server, but the same thing applies there that the full form is loaded on render, and then client-side managed from there.
